src/app/prt-staking/components/prt-section/prt-widget.tsx

```diff
@@ -63,10 +63,15 @@ export function PrtWidget({ token, onClose }: Props) {
     return ''
   }, [currentTab])
 
+  const isTradeButtonDisabled =
+    inputAmount.length === 0 || (currentTab === WidgetTab.STAKE && !canStake)
+
   const onClickTradeButton = useCallback(async () => {
+    if (isTradeButtonDisabled) return
     if (currentTab === WidgetTab.STAKE) {
       if (!isApproved) {
-        await onApprove()
+        const approved = await onApprove()
+        if (!approved) return
```

Comment:
`onApprove` has a `catch` already in it, so to avoid having to rewrite other code calling `onApprove` it's returning a bool denoting if it completed successfully